vector/src/main/java/im/vector/app/features/home/room/detail/timeline/action/MessageActionsViewModel.kt

```diff
@@ -362,6 +370,16 @@ class MessageActionsViewModel @AssistedInject constructor(
                 add(EventSharedAction.AddReaction(eventId))
             }
 
+            if (actionPermissions.canPinMessage && vectorPreferences.arePinnedMessagesEnabled()) {
+                val id: String = timelineEvent.root.eventId ?: return
+                val isPinned: Boolean = room?.stateService()?.isPinned(id) ?: return
```

Comment:
I would avoid using `return` here, this can lead to bugs.

you can just write:

```kotlin
if (actionPermissions.canPinMessage && vectorPreferences.arePinnedMessagesEnabled()) {
                val isPinned: Boolean = room?.stateService()?.isPinned(id).orFalse()
                if (isPinned) {
                    add(EventSharedAction.UnpinMessage(eventId))
                } else {
                    add(EventSharedAction.PinMessage(eventId))
                }
            }
```

(`eventId` is already declared)